be/src/storage/tablet_manager.cpp

```diff
@@ -856,25 +856,27 @@ Status TabletManager::start_trash_sweep() {
         }
     }
 
-    std::vector<TabletSharedPtr> tablets_to_check;
+    std::vector<DroppedTabletInfo> tablets_to_check;
     {
         std::shared_lock l(_shutdown_tablets_lock);
         tablets_to_check.reserve(_shutdown_tablets.size());
-        for (auto& [tablet_id, tablet_ptr] : _shutdown_tablets) {
-            tablets_to_check.emplace_back(tablet_ptr);
+        for (auto& [tablet_id, info] : _shutdown_tablets) {
+            tablets_to_check.emplace_back(info);
         }
     }
 
     std::vector<TTabletId> finished_tablets;
     finished_tablets.reserve(tablets_to_check.size());
 
-    for (const auto& tablet : tablets_to_check) {
+    for (const auto& info : tablets_to_check) {
+        auto& tablet = info.tablet;
         // The current thread has two references: _shutdown_tablets[i] and tablets_to_checks[i], check
         // if there is another thread has reference to the tablet, and if so, does not remove the tablet for now.
         if (tablet.use_count() > 2) {
             continue;
         }
 
+        bool remove_meta = false;
         TabletMeta tablet_meta;
         Status st = TabletMetaManager::get_tablet_meta(tablet->data_dir(), tablet->tablet_id(), tablet->schema_hash(),
```

Comment:
If you _remove_tablet_meta when flag == kDeleteFiles, how about the result status?

Reply:
The status would be NOT FOUND

Reply:
OK